### PR TITLE
Log Order book on TWAP tick

### PR DESCRIPTION
**Pull request by vitaliystoliarovcc**
Asana ticket: https://app.asana.com/0/1125859137800433/1200404383920919/f

- Modified the CSV Logger to allow create files with custom names
- Fixed closing of FS streams
- Log Order Book data on each TWAP tick

### CODE REVIEW

File: lib/host/events/log_algo_data.js [hunk 1]
```diff
@@ -23,19 +23,19 @@ module.exports = async (aoHost, gid, values) => {
...
 
   if (!initialized) {
     return
   }
 
-  algoLogWriter.write(values)
+  await algoLogWriter.write(filename || gid, values)
 
-  await updateState(instance, { algoLogWriter })
+  await updateState(instance, { algoLogWriter }) // TODO close CsvLogWriter on Host destroy
```
Comment (robertkowalski) [inline lib/host/events/log_algo_data.js:40]: still todo?

File: lib/twap/events/self_interval_tick.js [hunk 2]
```diff
@@ -99,6 +99,9 @@ const onSelfIntervalTick = async (instance = {}) => {
 
   const order = generateOrder(state, orderPrice)
   if (order) {
+    if (hasOBTarget(args) && lastBook) {
+      await emit('exec:log_algo_data', gid, lastBook.serialize(), `OB_for_oder_${order.cid}`)
```
Comment (robertkowalski) [inline lib/twap/events/self_interval_tick.js:103]: OB_for_order_

File: lib/twap/index.js [hunk 3]
```diff
@@ -51,6 +51,7 @@ const config = require('./config')
 const TWAP = defineAlgoOrder({
   id: 'bfx-twap',
   name: 'TWAP',
+  headersForLogFile: ['_', 'PRICE', 'TOTAL'],
```
Comment (robertkowalski) [inline lib/twap/index.js:54]: we can add ID here, this way people are not puzzling what `_` means.

ID, PRICE, TOTAL

File: lib/host/events/log_algo_data.js (hunk 1 above)
Comment (vitaliystoliarovcc) [reply lib/host/events/log_algo_data.js]: nope. removed

**Review by robertkowalski — APPROVED** [review]
lgtm

**Pull request merged**